src/gfx/linalg.hpp

```diff
@@ -14,6 +15,66 @@ inline void packFloat4x4(const float4x4 &mat, float *dst) {
   }
 }
 
+// compute the near and far intersections of the cube (stored in the x and y components) using the slab method
+// no intersection means vec.x > vec.y (really tNear > tFar)
+inline float2 intersectAABB(const float3 &rayOrigin, const float3 &rayDir, float3 boxMin, float3 boxMax) {
```

Comment:
The parameters qualifiers are inconsistent, some are passed by const reference and some by value, but they all have the same type?

Reply:
Fixed